comp/forwarder/defaultforwarder/internal/retry/http_transactions_serializer_test.go

```diff
@@ -112,6 +115,65 @@ func TestHTTPTransactionSerializerMissingAPIKey(t *testing.T) {
 	r.Equal(1, errorCount)
 }
 
+func TestHTTPTransactionSerializerUpdateAPIKey(t *testing.T) {
```

Comment:
Would it make sense to also test this scenario?

- Transaction is created with the old key. 
- Keys are updated.
- Transaction is being serialized.

Reply:
Urg... good thinking.

Reply:
I actually need to maintain a history of replacers, so we catch any outdated API keys. Performance is a concern, but hopefully API keys are updated rarely enough that it shouldn't be an issue..

Reply:
Keeping a history of API keys works, and I could be missing something, but I think it would be better to not rely on string replacement at all: keep the api key identity (index) in HTTPTransaction as it is created, serialize it in a dedicated `HTTPTransactionProto` field and re-set the correct key when reading. Support for API in the query string seems to have been removed from the code, so all that remains for the serializer to do is remove the api key header header.